static_src/components/app_container.jsx

```diff
@@ -120,11 +127,18 @@ export default class AppContainer extends React.Component {
   }
 
   render() {
-    let content = <h4 className="test-none_message">No app</h4>;
+    let loading = <div></div>;
+    let content = <div></div>;
 
     if (this.state.loading) {
-      content = <Loading text="Loading app now" />
-    } else if (this.state.app.name) {
+      loading = <Loading text="Loading app now" />;
+    }
+
+    if (this.state.empty) {
+      content = <h4 className="test-none_message">No app</h4>;
+    }
+
+    if (appReady(this.state.app)) {
```

Comment:
These are three mutually exclusive states right? I think it would be more clear if that was in `else if` blocks